api/redhatopenshift/Hypershift/hcpCluster-models.tsp

```diff
@@ -0,0 +1,559 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using Azure.Core;
+using Azure.ResourceManager;
+
+namespace Microsoft.RedHatOpenshift;
+
+/*
+* ===================================
+*   HyperShift cluster core resources
+* ===================================
+*/
+
+/** HyperShift cluster resource */
+model HcpOpenShiftClusterResource is TrackedResource<HcpOpenShiftClusterProperties> {
+  /** Name of HyperShift cluster */
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @minLength(3)
+  @maxLength(24)
+  @key("hcpOpenShiftClusterName") // sets the alternative name for the name propery
+  @path
+  @segment("hcpOpenShiftClusters")
+  name: string;
+
+  ...ManagedServiceIdentity;
+}
+
+// The NodePool needs to be TrackedResource for the following reasons:
+// - allow tags to be in place, which allows billing to use tags, important for workers
+// - allow nodepool to be tracked in the portal
+// - deleted when resource group is deleted
+// - allow cascade delete
+// more: https://armwiki.azurewebsites.net/rp_onboarding/tracked_vs_proxy_resources.html
+@parentResource(HcpOpenShiftClusterResource)
+model HcpOpenShiftClusterNodePoolResource is TrackedResource<NodePoolProperties> {
+  /** Name of HyperShift cluster */
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  @minLength(3)
+  @maxLength(24)
+  @key("nodePoolName")
+  @path
+  @segment("nodePools")
+  name: string;
+}
+
+/** HyperShift cluster properties */
+model HcpOpenShiftClusterProperties {
+  /** The status of the last operation. */
+  @visibility("read")
+  provisioningState?: ProvisioningState;
+
+  /** The cluster resouce specification. */
+  @visibility("create", "update")
+  spec: ClusterSpec;
+}
+
+/** The cluster resource specification */
+model ClusterSpec {
+  /** Version of the control plane components */
+  @visibility("create", "update")
+  version: VersionProfile;
+
+  /** Cluster DNS configuration */
+  dns?: DnsProfile;
+
+  /** Cluster network configuration */
+  @visibility("create")
+  network?: NetworkProfile;
+
+  /** Shows the cluster web console information */
+  @visibility("read")
+  console: ConsoleProfile;
+
+  /** Shows the cluster API server profile */
+  @visibility("read")
+  api: ApiProfile;
+
+  /** Enable FIPS mode for the cluster
+   * When set to true, `etcdEncryption` must be set to true
+  */
+  @visibility("create")
+  fips?: boolean = false;
+
+  /** Enables customer ETCD encryption, set during creation
+   * When set to true, `platform.etcdEncryptionSetId` must be set
+  */
+  @visibility("create")
+  etcdEncryption?: boolean = false;
```

Comment:
This field strikes me as redundant.  Could we either drop it or make it read-only and just say the presence of a non-empty `platform.etcdEncryptionSetId` in the PUT request implies encryption is enabled?

Reply:
The reason we use this is to provide fast-fail validation. For example, if you set FIPS=true, we force etcdEncryption to also be true, which forces the request to fail if encryptionSetId is empty. There might be other use cases (at least we have some others in CS why we do it this way) that may or may not apply.

Reply:
Presumably enforcing this consistency will be part of the RP's static validation.  Still seems to me this field could be read-only; the static validation can check if `etcdEncryptionSetId` is non-empty as easily as whether `etcdEncryption` is true.  What you're describing wrt FIPS sound like an implementation detail.